src/Microsoft.AspNetCore.Server.Kestrel/Http/Frame.cs

```diff
@@ -851,7 +859,10 @@ public bool TakeMessageHeaders(SocketInput input, FrameRequestHeaders requestHea
                 while (!scan.IsEnd)
                 {
                     var beginName = scan;
-                    scan.Seek(ref _vectorColons, ref _vectorCRs);
+                    if (scan.Seek(ref _vectorColons, ref _vectorCRs) == -1)
+                    {
+                        return false;
```

Comment:
In this case, isn't this already taken care of by the following?

    if (chFirst == -1 || chSecond == -1)
    {
        return false;
    }